backend/src/schema/types/type/Badge.gql

```diff
@@ -1,6 +1,5 @@
 type Badge {
   id: ID!
-  key: String!
   type: BadgeType!
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-11T14:43:59Z" title="Thursday, July 11th 2019, 4:43:59 pm +02:00">Jul 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,6 +1,5 @@
 type Badge {
-  id: ID!
-  key: String!
+  key: ID!
```


does this change mean that uniqueness and immutability are introduced?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-11T19:54:27Z" title="Thursday, July 11th 2019, 9:54:27 pm +02:00">Jul 11, 2019</time>_

---


The `ID!` means the server will crash if it ever would return `null` for `key`. What do you mean by immutability? We can still update it if we really wanted to (even though that's a silly idea).

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-12T07:57:58Z" title="Friday, July 12th 2019, 9:57:58 am +02:00">Jul 12, 2019</time>_

---


Then this Badge nodes will have no `id` property.
I have the impression that all nodes should have a property `id` to identify them … !?

If it's the identifier, then name it always like this, please. Otherwise that can lead to confusion.
And in Tag nodes we name it `id` as well.